subxt/src/tx/tx_progress.rs

```diff
@@ -95,10 +98,10 @@ where
     /// **Note:** consumes `self`. If you'd like to perform multiple actions as the state of the
     /// transaction progresses, use [`TxProgress::next_item()`] instead.
     ///
-    /// **Note:** transaction statuses like `Invalid` and `Usurped` are ignored, because while they
-    /// may well indicate with some probability that the transaction will not make it into a block,
-    /// there is no guarantee that this is true. Thus, we prefer to "play it safe" here. Use the lower
-    /// level [`TxProgress::next_item()`] API if you'd like to handle these statuses yourself.
+    /// **Note:** transaction statuses like `Invalid`/`Usurped`/`Dropped` indicate with some
+    /// probability that the transaction will not make it into a block but there is no guarantee
+    /// that this is true. In those cases the stream is closed however, so you currently have no
+    /// way to find out if they finally made it into a block or not.
```

Comment:
it's possible to create another subscription and watch the extrinsic again if any of the transaction statuses wasn't finalized.

For example

```rust
let xt = create_xt()
submit(xt.clone).await.unwrap()
loop {
   match watch_until_in_finalized(xt).await? {
       TxStatus::Finalized(_) => break Ok(()),
       other => {
           log::warning!("Watch extrinsic {xt} failed: {other}, retrying again);
       }
   }
}
```

Reply:
for sure complicated but possible

Reply:
Ah okay, I did not know this, should I add your code example to the doc comment?

Reply:
That code doesn't really work for the `TxProgress` as it wraps subscription and extrinsic.
So you can just modify your comment `In those cases the stream is closed however, so you currently have no way to find out` to something like `In those cases you have to re-subscribe to the extrinsic/create a new TxProgress repeatedly until the extrinsic is finalized/finalizetimeout `

Reply:
I'm not sure whether we should provide an API for this, what do you guys think?

Reply:
Ooh hold on; I didn't think that it was possible to "re subscribe" to a transaction again!

Reply:
> I'm not sure whether we should provide an API for this, what do you guys think?

You mentioned above that "the old RPC API will be replaced soon", so does that mean that also our stream handling will change? If that is the case, will the logic of this API need to change? If not it is certainly a good Idea to have an API for resubscribing right now, something like `tx_progress.wait_until_finalized_until_timeout()` that does the resubscribing under the hood.